### PR TITLE
Add runtime metadata

### PR DESCRIPTION
**Pull request by cournape**
This PR implements a function `runtime_metadata_factory` to parse metadata from runtime files.

This incidentally defines the new format.

This implementation is incomplete in the sense that I'd like to add jsonschema-based validation of the various metadata formats, but I thought I would do that in a separate PR.

### CODE REVIEW

File: doc/source/user/quickstart.rst [hunk 1]
```diff
@@ -8,11 +8,24 @@ started with okonomiyaki.
...
+<okonomiyaki.file_formats.runtime_metadata_factory>` function::
+
+    >>> from okonomiyaki.file_formats import runtime_metadata_factory
+
+To query metadata of an existing runtime::
+
+    >>> metadata = runtime_metadata_factory("python-cpython-2.7.10-1-rh5_x86_64.runtime")
+    >>> print(metadata.implementation)
+
+Every metadata instance share the attributes of the class RuntimeMetadataV1,
```
Comment (itziakos) [inline doc/source/user/quickstart.rst]: `share` -> `shares`

File: okonomiyaki/file_formats/runtime.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+    if isinstance(path_or_file, six.string_types):
+        with zipfile2.ZipFile(path_or_file) as zp:
+            metadata = _read_runtime_metadata_json(zp)
+    else:
+        metadata = _read_runtime_metadata_json(path_or_file)
+
+    json_dict = json.loads(metadata)
+    key = _factory_key_from_metadata(json_dict)
+    klass = _METADATA_KLASS_FACTORY.get(key)
```
Comment (itziakos) [inline okonomiyaki/file_formats/runtime.py]: I assume that at a later version the class registry will be an optional argument correct?

File: okonomiyaki/file_formats/runtime.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+from okonomiyaki.versions import MetadataVersion, SemanticVersion
+
+
+RuntimeVersion = SemanticVersion
+
+_METADATA_ARCNAME = "metadata/runtime.json"
+
+
+@attributes
+class RuntimeMetadataV1(object):
```
Comment (itziakos) [inline okonomiyaki/file_formats/runtime.py]: is the `V1` in the name necessary?

File: okonomiyaki/file_formats/runtime.py (hunk 2 above)
Comment (cournape) [inline okonomiyaki/file_formats/runtime.py]: No, since the mapping will always be static.

File: okonomiyaki/file_formats/runtime.py (hunk 3 above)
Comment (cournape) [inline okonomiyaki/file_formats/runtime.py]: Yes, otherewise how would you differentiate with say "1.1" metadata which would add some more attributes ?

**Comment by itziakos** [comment]
a few comments/questions apart from that  :+1:

**Pull request merged**